### DISCUSSION TITLE
Is Redis worth adding in compose / use cases?

### DISCUSSION THREAD

**Discussion by audioscavenger** [LycheeOrg/Lychee#2322] [Q&A] [answered]
I read all the documentation related to caching and am not sure what I am missing. Spent some time setting all up and all seems to work.
* doc mentions the need for phpredis
* container is provided by Linuxserver.io
* lychee starts with no error, meaning it's working with Redis
* INFO keyspace returns a database, with 10 keys, created by lychee
* I also executed `php /app/www/artisan config:cache` but that may have nothing to do with Redis
* no errors in lychee logs

How do we test if there is any performance gain? I may be naive but I don't see any improvements, after navigating or uploading.
Is the point of Redis about better handling of tons of queries/uploads by many users?

thank you for any insight

**Comment by ildyria** [comment, marked as answer]
Redis is currently not supported by Lychee, as in, yes you can add it, but it does not have an implementation.
What would be the benefits once implemented ?

- caching requests which are often called when being a visitor.
- caching some specific heavy SQL queries.

> I also executed php /app/www/artisan config:cache but that may have nothing to do with Redis

This has nothing to do with Redis. `config:cache` pretty much creates a cache of all the php files `config/` with all the logic in it removed. It pretty much creates an single return array with all data populated. As a result, changes in your `.env` file no longer have effects.

**TL;DR: Not yet, but soon<sup>TM</sup>**
